src/compiler/scanner.ts

```diff
@@ -1664,6 +1671,72 @@ namespace ts {
             return token;
         }
 
+        function scanJSDocToken(): SyntaxKind {
+            startPos = pos;
+
+            // Eat leading whitespace
+            while (pos < end) {
+                const ch = text.charCodeAt(pos);
+                if (isWhiteSpace(ch)) {
+                    pos++;
+                }
+                else {
+                    break;
+                }
+            }
+            tokenPos = pos;
+
+            let identifierStarted = false;
+            while (pos < end) {
+                const ch = text.charCodeAt(pos);
+                if (identifierStarted) {
+                    if (!isIdentifierPart(ch, ScriptTarget.Latest)) {
+                        return token = SyntaxKind.Identifier;
+                    }
+                }
+                else {
+                    if (ch === CharacterCodes.at) {
+                        return pos += 1, token = SyntaxKind.AtToken;
+                    }
+                    else if (isLineBreak(ch)) {
+                        return pos += 1, token = SyntaxKind.NewLineTrivia;
+                    }
+                    else if (ch === CharacterCodes.asterisk) {
+                        return pos += 1, token = SyntaxKind.AsteriskToken;
+                    }
+                    else if (ch === CharacterCodes.openBrace) {
+                        return pos += 1, token = SyntaxKind.OpenBraceToken;
+                    }
+                    else if (ch === CharacterCodes.closeBrace) {
+                        return pos += 1, token = SyntaxKind.CloseBraceToken;
+                    }
+                    else if (ch === CharacterCodes.openBracket) {
+                        return pos += 1, token = SyntaxKind.OpenBracketToken;
+                    }
+                    else if (ch === CharacterCodes.closeBracket) {
+                        return pos += 1, token = SyntaxKind.CloseBracketToken;
+                    }
+                    else if (ch === CharacterCodes.equals) {
+                        return pos += 1, token = SyntaxKind.EqualsToken;
+                    }
+                    else if (ch === CharacterCodes.comma) {
+                        return pos += 1, token = SyntaxKind.CommaToken;
+                    }
+                    else if (isWhiteSpace(ch)) {
+                        // Keep going

```

Comment:
This should never happen though right? As you've eaten the preceding white space, and there's nothing in here where you would consume white space inside a token.